hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersCall.java

```diff
@@ -0,0 +1,255 @@
+        return gasOnly(successResult(encodedStatus(status), 0L));
+    }
+
+    private boolean executionIsNotSupported() {
+        return Arrays.equals(selector, CRYPTO_TRANSFER_V2.selector())
+                && !configuration.getConfigData(ContractsConfig.class).precompileAtomicCryptoTransferEnabled();
+    }
+
+    /**
+     * Indicates if the given call attempt is for {@link ClassicTransfersCall}.
+     *
+     * @param attempt the attempt to check
+     * @return {@code true} if the given {@code selector} is a selector for {@link ClassicTransfersCall}
+     */
+    public static boolean matches(@NonNull final HtsCallAttempt attempt) {
+        requireNonNull(attempt);
+        return !attempt.isTokenRedirect()
+                && (Arrays.equals(attempt.selector(), CRYPTO_TRANSFER.selector())
+                        || Arrays.equals(attempt.selector(), CRYPTO_TRANSFER_V2.selector())
+                        || Arrays.equals(attempt.selector(), TRANSFER_TOKENS.selector())
+                        || Arrays.equals(attempt.selector(), TRANSFER_TOKEN.selector())
+                        || Arrays.equals(attempt.selector(), TRANSFER_NFTS.selector())
+                        || Arrays.equals(attempt.selector(), TRANSFER_NFT.selector())
+                        || Arrays.equals(attempt.selector(), HRC_TRANSFER_FROM.selector())
+                        || Arrays.equals(attempt.selector(), HRC_TRANSFER_NFT_FROM.selector()));
+    }
+
+    /**
+     * Creates a {@link ClassicTransfersCall} from the given {@code attempt} and {@code senderAddress}.
+     *
+     * @param attempt         the attempt to create a {@link ClassicTransfersCall} from
+     * @param sender          the address of the sender
+     * @param onlyDelegatable whether the sender needs delegatable contract keys
+     * @return a {@link ClassicTransfersCall} if the given {@code attempt} is a valid {@link ClassicTransfersCall}, otherwise {@code null}
+     */
+    public static ClassicTransfersCall from(
+            @NonNull final HtsCallAttempt attempt,
+            @NonNull final org.hyperledger.besu.datatypes.Address sender,
+            final boolean onlyDelegatable) {
+        requireNonNull(attempt);
+        requireNonNull(sender);
+        final var selector = attempt.selector();
+        return new ClassicTransfersCall(
+                attempt.enhancement(),
+                selector,
+                attempt.addressIdConverter().convert(asHeadlongAddress(sender.toArrayUnsafe())),
+                nominalBodyFor(attempt),
+                attempt.configuration(),
+                isLegacyCall(selector) ? APPROVAL_SWITCH_HELPER : null,
+                attempt.verificationStrategies()
+                        .activatingOnlyContractKeysFor(sender, onlyDelegatable, attempt.nativeOperations()),
+                SYSTEM_ACCOUNT_CREDIT_SCREEN);
+    }
+
+    private static boolean isLegacyCall(@NonNull final byte[] selector) {
```

Comment:
I'm not sure if "legacy" is the best descriptor for this method.

Reply:
Changed to `isClassicCall()`